apps/emqx_management/test/emqx_mgmt_api_status_SUITE.erl

```diff
@@ -30,8 +36,82 @@ init_per_suite(Config) ->
 end_per_suite(_) ->
     emqx_mgmt_api_test_util:end_suite().
 
-t_status(_Config) ->
-    Path = emqx_mgmt_api_test_util:api_path_without_base_path(["/status"]),
-    Status = io_lib:format("Node ~ts is ~ts~nemqx is ~ts", [node(), started, running]),
-    {ok, Status} = emqx_mgmt_api_test_util:request_api(get, Path),
+init_per_testcase(t_status_not_ok, Config) ->
+    ok = application:stop(emqx),
+    Config;
+init_per_testcase(_TestCase, Config) ->
+    Config.
+
+end_per_testcase(t_status_not_ok, _Config) ->
+    {ok, _} = application:ensure_all_started(emqx),
+    ok;
+end_per_testcase(_TestCase, _Config) ->
+    ok.
+
+%%---------------------------------------------------------------------------------------
+%% Helper fns
+%%---------------------------------------------------------------------------------------
+
+do_request(Opts) ->
```

Comment:
Out of scope of this PR but have you considered using [katt](https://github.com/for-GET/katt) for testing APIs?

In your case you would just have to create scenario file like
```
GET /status
< 200
emqx is running
```
For the success case and similar for the failure. Rest is handled by a generic framework.

Reply:
Correct, it's meant as black box testing. But it's still ct. You can still setup things the way you need in your test case and then run the matching scenario. Just it doesn't get as generic anymore if you need one scenario per test. 

Often the desired state can be set by doing preparation calls from within the scenario though. I don't know if that's possible for what you have in mind of course.

Reply:
I see, thanks for the explanation!  I'll read it better and try to understand it.  :smiley_cat: 

Seem like converting the existing tests to start using `katt` could be done in separate PRs, what do you think?

Reply:
I'd agree, this needs some proper planning since you typically want some scaffolding around katt to setup things, run and evaluate results (including errors) in a more streamlined way, matching our project's needs. In other word, a small testlib on our side that encapsulates it.